cmd/rpm/main.go

```diff
@@ -0,0 +1,570 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/aws/aws-sdk-go-v2/service/s3/types"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+const (
+	newRepoPath    = "/tmp/new_repo"
+	oldRepoPath    = "/tmp/old_repo"
+	mergedRepoPath = "/tmp/merged_repo"
+	version        = "development"
+)
+
+type RpmCmdOpts struct {
+	Bucket       string
+	Prefix       string
+	Visibility   string
+	AwsAccessKey string
+	AwsSecretKey string
+	AwsRegion    string
+	Sign         bool
+	SignPass     string
+	RpmFiles     []string
+	Rebuild      bool
+}
+
+var rpmCmdOpts RpmCmdOpts
+
+func signRepo(password, repoPath string) error {
+	repomdPath := fmt.Sprintf("%s/repodata/repomd.xml", repoPath)
+
+	if password != "" {
+		command := fmt.Sprintf(`
+expect -c '
+set timeout 60
+spawn gpg --pinentry-mode loopback --force-v3-sigs --verbose --detach-sign --armor %s
+expect -re "Enter passphrase.*"
+send -- "%s\r"
+expect eof
+lassign [wait] _ _ _ code
+exit $code
+'
```

Comment:
it feels odd to have golang running a shell that runs expect seems... messy. Is there a golang native way to do this?